### PR TITLE
feat: implement Chen+24 particle spray model

### PR DESCRIPTION
**Pull request by ybillchen**
This branch implements the new Chen+24 ([arXiv:2408.01496](https://arxiv.org/abs/2408.01496)) model for stream modeling. A [notebook](https://github.com/ybillchen/particle_spray) demonstrating how to use the model is available.

-Bill Chen

### CODE REVIEW

**Review by nstarman — COMMENTED** [review]
Thanks @ybillchen! This looks great.
I was looking through the provided [notebook](https://github.com/ybillchen/particle_spray/blob/main/galax_stream.ipynb) and noted the support is partial because we have not yet implemented the progenitors' potential. If you could add a `warnings.warn` about the partial support when the DF is instantiated then I think this would be ready to merge.

**Comment by ybillchen** [comment]
Sounds great! Thanks for reviewing the code and I have added the warning.

File: src/galax/dynamics/_dynamics/mockstream/df/chen24.py [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+    """
+    return qnp.sqrt(2*potential.constants["G"]*prog_mass/rad)
+
+
+#####################################################################
+# TODO: below is copied from fardal15.py
+#       Change accordingly when moved to a more general location.
+
+#####################################################################
+# TODO: move this to a more general location.
```
Comment (nstarman) [inline src/galax/dynamics/_dynamics/mockstream/df/chen24.py:141]: Would you mind just importing directly from fardal.py?
I'm prepping a PR to follow this one cleaning up these functions and moving them to a more sensible location.

Comment (ybillchen) [reply src/galax/dynamics/_dynamics/mockstream/df/chen24.py]: Sure!

**Comment by nstarman** [comment]
@ybillchen you might have to rebase, not merge. I'm seeing a lot of old commits getting picked up.

**Comment by ybillchen** [comment]
> @ybillchen you might have to rebase, not merge. I'm seeing a lot of old commits getting picked up.

Oops! My bad. Switched to rebasing!

**Review by nstarman — APPROVED** [review]
Thanks @ybillchen for the great addition!
As we continue to develop the documentation and test suite would you be willing to contribute plots / tests ?
Also, let's get another approval. Ping @adrn @jnibauer :)

**Comment by ybillchen** [comment]
> Thanks @ybillchen for the great addition!
> 
> As we continue to develop the documentation and test suite would you be willing to contribute plots / tests ?
> 
> Also, let's get another approval. Ping @adrn @jnibauer :)

Sounds good! Very happy to contribute to this great project

**Comment by jnibauer** [comment]
all looks great to me!

**Comment by nstarman** [comment]
Excellent. Thanks @ybillchen !

**Pull request merged**